src/Kunstmaan/FormBundle/Entity/FormSubmissionFieldTypes/FileFormSubmissionField.php

```diff
@@ -95,14 +94,26 @@ public function upload($uploadDir, $webDir)
         $uuid = uniqid();
         $this->setUuid($uuid);
 
-        // move takes the target directory and then the target filename to move to
-        $this->file->move(sprintf('%s/%s', $uploadDir, $uuid), $safeFileName);
+        if ($fileSystem) {
+            $url = \sprintf('%s%s/%s', $webDir, $uuid, $safeFileName);
+            $content = \file_get_contents($this->file->getPathname());
+            $fileSystem->write($url, $content);
+        } else {
+            @trigger_error(
+                    'Not passing the filesystem as the third argument of upload is deprecated since KunstmaanMediaBundle 5.7 and will be required in KunstmaanMediaBundle 6.0.',
```

Comment:
```suggestion
                    sprintf('Not passing a valid value for the "$filesystem" argument of "%s" is deprecated since KunstmaanFormBundle 5.8 and will be required in KunstmaanFormBundle 6.0.', __METHOD__),
```

Reply:
Done